### ISSUE TITLE
New quest: tag bin=* at bus stops

### ISSUE THREAD

**Issue by tiptoptom** [streetcomplete/StreetComplete#2760]
### General
Add tagging to indicate if a bus stop has a waste basket: [bin=*](https://wiki.openstreetmap.org/wiki/Key:bin)

Affected tag(s) to be modified/added: bin=* on nodes where highway=bus_stop and bin=* is not set
Question asked: **Does this bus stop have a waste basket?**
Valid responses: Yes | No

### Checklist
Checklist for quest suggestions (see guidelines):
- [x] 🚧 To be added tag is established and has a useful purpose
- [x] 🤔 Any answer the user can give must have an equivalent tagging (Quest should not reappear to other users when solved by one)
- [x] 🐿️ Easily answerable by everyone from the outside but a survey is necessary 
- [x] 💤 Not an overwhelming percentage of elements have the same answer (No spam)
- [x] 🕓 Applies to a reasonable number of elements (Worth the effort)

### Ideas for implementation

**Element selection:** 

**Metadata needed:** 

**Proposed GUI:**

**Comment by matkoniecz** [comment]
See #200

**Comment by tiptoptom** [comment]
> See #200

Am I right that the main counterargument was that trash cans are of no use to anyone?

**Comment by matkoniecz** [comment]
Yes, even "here is a precise location of trash can" is kind of dubious.

"this bus stop has nearby trash can" seems even more dubious.

**Comment by mnalis** [comment]
I'm genuinely interested why people find waste baskets useless? Maybe many are car-based, but for pedestrian (and cyclist) I  find then quite useful.

For example fast food / bakery grease paper (here it's quite common you'd buy those and eat them on the go), or used paper towels / handkerchiefs, or (over)used disposable COVID-19 medicine mask, used tickets, cigarette butts, etc.? 
What people do when they need to dispose of such small-amount trash, throw it on the street (unfortunately way more common then I'd like in my city), or carry it for next hour with themselves until they reach their destination? 

Or do your cities have them every dozen meters so it's not an issue? Here unfortunately (outside of strict city centre) you might be usually required to detour  (or find alternative route) at least several hundreds meters, to up to few kilometers to find one. I myself hate littering, so I'd detour and throw it in nearest wastebasket. 

And bus stops are are one of more likely places to find them (although there is no guarantee - same as bench/shelter, so this would be useful). My limited visits to other countries also seem to strengthen that theory (wastebaskets often found at bus stops).

**So I'd find that quest quite useful**. At least as useful as if bus stop has a shelter, and more than if bus shelter has a bench. (even it has  bench, and you are very tired, there is no guarantee that it will be available - as likely it will not be). 

And for shelter, most common use is also unrelated to bus/tram stop itself (one is quite unlikely to walk several kilometers in rain to another bus stop in order to use it's shelter!) - you simply want to find nearest shelter (or nearest waste basket). The extra-precise location is not as important to me (I don't really care if it is 10 meters here or there)

(And definitely seems to me waay more useful that "what is type of roof", or "what was original purpose the building was built for" existing quests)

**Comment by matkoniecz** [comment]
Idea of checking for nearby trash can is something that I never really considered, so thanks for sharing that.

> I'm genuinely interested why people find waste baskets useless? Maybe many are car-based, but for pedestrian (and cyclist) I find then quite useful.

> For example fast food / bakery grease paper (here it's quite common you'd buy those and eat them on the go), or used paper towels / handkerchiefs, or (over)used disposable COVID-19 medicine mask, used tickets, cigarette butts, etc.?

My comment was about usefulness of mapping it, not about usefulness of trash cans itself.

> Or do your cities have them every dozen meters so it's not an issue?

In my city (Kraków) I would expect one to be within sight. In rare cases when nothing was within a reasonable range I asked local government to place additional one, what was realised.

> Here unfortunately (outside of strict city centre) you might be usually required to detour (or find alternative route) at least several hundreds meters, to up to few kilometers to find one. 

On hiking trips I would carry it in backpack/pocked (withing bag/trash usable as a bag) - I would not check map for nearby trash cans to detour.

> what was original purpose the building was built for

Here primary purpose is to gather info for address quest.

> what is type of roof

I am not going to defend this one (or backrest quest) :) But roof shape is widely used in 3D renderings.

**Comment by mnalis** [comment]
> Idea of checking for nearby trash can is something that I never really considered, so thanks for sharing that.

I'd even say after "favorite" / "address" / "food" / "drink" / "medical", it's one of my more  common "find amenity" target.

> > Or do your cities have them every dozen meters so it's not an issue?
> 
> In my city (Kraków) I would expect one to be within sight. In rare cases when nothing was within a reasonable range I asked local government to place additional one, what was realised.

Yes, that would explain it. The same as someone in Amsterdam might be confused by silliness of "is this cycleway separated from footway" type of question, I guess. 

Still, some countries are way more lucky there than others. 

Wastebaskets are nowhere near as available here, and our level of asking local government is more on the level _"please put bollards so people can't park cars on sidewalk on this 100m busy street that schoolchildren need to walk every day"_ (and even that requires petition of hundreds individuals and is nowhere near certain it will actually be done `</rant>`)

> > Here unfortunately (outside of strict city centre) you might be usually required to detour (or find alternative route) at least several hundreds meters, to up to few kilometers to find one.
> 
> On hiking trips I would carry it in backpack/pocked (withing bag/trash usable as a bag) - I would not check map for nearby trash cans to detour.

On hiking trips, so would I put it in trashbag in backpack, but for regular moving in the city between places, not really (as I won't have backpack and I usually don't have that big pockets even if I wanted to - and compressed greasy food remains would put me very much on "I don't want to" side, as that would be quite hard to grease-proof)

> I am not going to defend this one (or backrest quest) :) But roof shape is widely used in 3D renderings.

I'm sure someone will find a use case for anything (hey, some might like to "[visit only power poles which are made of steel instead of wood or concrete"](https://github.com/streetcomplete/StreetComplete/issues/1227) - I'm not judging!) but I was just trying to hint that mapping / finding wastebaskets actually is quite useful for some people.

**Comment by matkoniecz** [comment]
sorry for offtopic but I wanted to correct possibly misleading impression:

> Wastebaskets are nowhere near as available here, and our level of asking local government is more on the level "please put bollards so people can't park cars on sidewalk on this 100m busy street that schoolchildren need to walk every day" (and even that requires petition of hundreds individuals and is nowhere near certain it will actually be done </rant>)

Well, doing things that would impact cars negatively is kind of tricky here (though happens sometimes, especially in recent years). City government placed thousands of parking sites blocking sidewalks. Despite that it was simply illegal. And removed them only after some official were officially threatened by fines/imprisonment for breaking law.

https://dziennikpolski24.pl/zbyt-malo-miejsca-dla-pieszych-na-chodnikach-w-strefie-parkowania/ar/9160856

![screen18](https://user-images.githubusercontent.com/899988/115137824-9a223080-a028-11eb-9f97-b57b74308c73.png)

**Comment by westnordost** [comment]
My comment about usefulness and that there are many more important quests yet to be implemented is 4 years old (version 1.0), back then, there were just 20 quest types in total.

So, I guess this could be done.

**Comment by DerDings** [comment]
I don't see much value in `bin=yes` on bus stops in urban areas. There are already a lot of quests for bus stops. While they're not spammy individually, they become so when enabled all together. 
When having two bus stops nearby, I can imagine that one would like to be able to choose the one which has a shelter, or the one which has a bench. Also a PT router may prefer stops with a shelter and bench for changing. In a nutshell, those are important _properties_ of bus stops.
But I cannot imagine many people are choosing to use a certain bus stop just because it has got a bin. A bin is not an important _bus stop property_.

For those people just looking for the location of the next bin itself, it is more useful to map the actual location of the bin rather than unprecisely on an object away from the exact location. 

[#200 (Comment)](https://github.com/streetcomplete/StreetComplete/issues/200#issuecomment-299949220):
> I used Humanitarian layer with waste baskets as map illustration in petition to local government (obviously, the petition was about waste baskets missing in one particular region).

Determining the density of trash cans, or finding spots where no trash can is nearby to set up a new one, are use cases I imagine to be useful. This is more difficult if trash cans are mapped twice, as a node and as `bin=yes` on objects many meters away.

However, it may be useful in rural areas, where the exact location of the bin really doesn't matter that much, or in places where the trash cans themselves aren't mapped. But I think all bus stops with an object tagged `amenity=waste_basket access != private|no` or `bin=yes access != private|no` nearby should be excluded. I'd suggest a buffer of 35 meters (a medium sized bus stop: two [mono-articulated buses](https://en.wikipedia.org/wiki/Articulated_bus) are 2*18=36 m). The `highway=bus_stop` node is often placed near the front door of the bus (where blind people and non-locals get on the bus), while the waste basket may be placed at the end (to be out of way when getting on/off the bus, regardless where the used vehicle has it's doors placed).

**Comment by harry-wg** [comment]
I'm finding London is getting over-run by map notes indicating mappers have been unable to answer the quest because the bus stop has a bin, but it is separate. Most of London's 19 thousand of bus stops have a shelter, which does not have a bin built in to it, But in almost every case, there is a bin within a few feet of the stop.

In the opening post it states that the valid responses are "yes" and "no". https://wiki.openstreetmap.org/wiki/StreetComplete/Quests also documents the values as only yes or no. But at https://wiki.openstreetmap.org/wiki/Tag:highway%3Dbus_stop it is stated that the valid values are yes / no / separate.

This seems to be a very new phenomenon and it would greatly help preserve the sanity of the London editors if it could be possible for those using StreetComplete to add a tag of "separate" (without leaving a note),

**Comment by matkoniecz** [comment]
`bin=yes` is correct for cases where there is a separately mapped waste basket.

**Comment by matkoniecz** [comment]
I edited https://wiki.openstreetmap.org/wiki/Tag:highway%3Dbus_stop#Physical_attributes (if you think that it is a mistake - I would be happy discuss it at OSM Wiki or tagging mailing list, please do not discuss it here as issue tracker of a specific project is a poor place to discuss a tagging issues).

But feel free to mention a potential discussion.

**Comment by harry-wg** [comment]
I'm not personally worried whether the values are yes/no or yes/no/separate. But if they are to be only yes/no, can you write something in the quest to make it clear that they should answer **yes** for a separate bin, and **not** flood the map with pointless and unnecessary notes to say so.

**Comment by matkoniecz** [comment]
Is it just this specific mapper or problem more general? 

I looked for notes using https://github.com/streetcomplete/StreetComplete/blob/master/CONTRIBUTING.md#solving-notes and "flooding" appears to be accurate and should be resolved.

I solved some notes like https://www.openstreetmap.org/note/2787617 and plan to process all of them (but I want to wait a bit and give this mapper chance to response, maybe they have valid reasons for rejecting this)

Either way, thanks for bringing it to attention - note flooding like this should not happen (except cases where data is truly invalid).

**Comment by peternewman** [comment]
I've opened quite a few (but not all) of these. Looking at some others, there's also for example:
https://www.openstreetmap.org/note/2743961

There are also quite a few people finding the stop is gone, but also more bafflingly some writing no in the note, rather than pressing no, although I don't regularly look at other people's notes, so I don't know how that compares to normal.

I think the simple solution is to add `separate` as another answer to SC like we do with pavements and ramps (doing it for benches would be good too).

The image in the wiki article is clearly yes:
https://wiki.openstreetmap.org/wiki/File:Berkefeldweg_Bushaltestelle_mit_gl%C3%A4sernem,_Graffiti_-verschmiertem_Wartehaus,_M%C3%BClleimer_und_Sitzbank_in_der_Wittinger_Stra%C3%9Fe_in_Celle.jpg

But from my understanding of:
https://wiki.openstreetmap.org/wiki/One_feature,_one_OSM_element

If the bin isn't part of the bus stop, it should be tagged with it's own node and marked as separate on the stop. In the same way as we'd do (and I've done) for the rare occasions there are bus stops without benches but with one nearby (rather than part of a bus stop shelter). Otherwise a bit like some of the lit quests, you're getting into a vagueness of how close does a bin need to be to a bus stop to be yes, 1m, 5m, 10m?

> Also, with SC it is impossible to check whether there is a separately mapped waste basket

We could change SC to not ask if there is a waste basket nearby (or to tag separate). However tagging separate (even if there isn't a nearby bin on the map), would allow other QA tools to then flag up that there are items missing nearby which could be added, thereby improving the map further.

Also if you wanted to use it for something like https://www.fixmystreet.com/ I suspect the bus stop bins are managed by TfL but the normal bins would be looked after by the local council, so faults would need to be directed to different people.

TLDR: If I'm going to go to the effort of collecting the data, I'd like to do it in as accurate a way as possible. Tagging anything but separate if there's an adjacent bin results in the potential for double-counting and other forms of duff info.

**Comment by andrewharvey** [comment]
> But from my understanding of:
https://wiki.openstreetmap.org/wiki/One_feature,_one_OSM_element
>
> If the bin isn't part of the bus stop, it should be tagged with it's own node and marked as separate on the stop.

I come to a different conclusion, that per one feature one osm element, bin=yes on a highway=bus_stop without a nearby amenity=waste_basket is incomplete data and eventually someone should map add add the amenity=waste_basket node.

This isn't double counting since bin=yes is just saying that this bus stop has a waste basket available, but it's not saying the bus stop is the waste basket.

**Comment by peternewman** [comment]
> I come to a different conclusion, that per one feature one osm element, bin=yes on a highway=bus_stop without a nearby amenity=waste_basket is incomplete data and eventually someone should map and add the amenity=waste_basket node.

So if the bin is on the post itself, do you add the amenity=waste_basket tag to the highway=bus_stop node itself? Or put another node practically on top of it?

> This isn't double counting since bin=yes is just saying that this bus stop has a waste basket available, but it's not saying the bus stop is the waste basket.

The wiki currently says "'yes' if there is a trash can at the bus stop (additionally it may or may not be mapped separately with amenity=waste_basket).". So if there's `highway=bus_stop` and `bin=yes` and a nearby `amenity=waste_basket` is that one or two bins?

At least if it was `highway=bus_stop` and `bin=separate` and a nearby `amenity=waste_basket` I can tune my heuristic for what counts as nearby to decide if it's 1 or 2 rather than just having to completely guess (and crucially if there isn't a `amenity=waste_basket` I know the answer is 1).

**Comment by matkoniecz** [comment]
> I think the simple solution is to add separate as another answer to SC like we do with pavements and ramps (doing it for benches would be good too).

Note that SC mapper is unable to distinguish between `yes` and `separate` as waste baskets are not shown in SC.

And there is one more fundamental issues - `separate` is a new, unpopular and barely used tag. See http://taghistory.raifer.tech/#***/bin/yes&***/bin/no&***/bin/separate

![screen03](https://user-images.githubusercontent.com/899988/129382103-1a94320f-55f1-4339-a581-96f5f299391f.png)

before promoting such tags in SC a wider community should be asked (though as mapper is unable to distinguish between `yes` and `separate` in SC it is not useful to do it)

> So if the bin is on the post itself, do you add the amenity=waste_basket tag to the highway=bus_stop node itself? Or put another node practically on top of it?

Neither as I see no value in mapping this, but in such case I would put node next to it.

> and crucially if there isn't a amenity=waste_basket

In such case `bin=separate` is simply wrong (even if that makes counting for one specific scenario easier)

>  https://wiki.openstreetmap.org/wiki/One_feature,_one_OSM_element

`bin=yes` is not marking a waste basket. It marks that something has associated waste basket. In particular, multiple objects with `bin=yes` may be associated with one waste basket.

Similarly to `bridge=yes` on road (mapping as a property) and `man_made=bridge` (maps a bridge).

Or `embankment=yes` on road or railway vs mapped `man_made=embankment`.

> guess

Note that it will be never eliminated - even if `bin=separate` will become accepted you will never know whether `yes` is in meaning "and separate waste basket object may be present" or in "and separate waste basket object is not present".

While adding `=separate` complicates mapping. Without a real benefit. Similarly how people introduced `bicycle=dismount` (duplicating already used `bicycle=no` with the same meaning) and years later we have two tags with the same meaning that still confuse people.

**Comment by peternewman** [comment]
> Note that SC mapper is unable to distinguish between `yes` and `separate` as waste baskets are not shown in SC.

A slightly tweaked version of #2354 would fix that (presuming we don't want to change the map theme).

Also thanks to the existence check being such a high priority, that's often not the case either, as you've potentially just marked any mapped bins as being present.
 
> And there is one more fundamental issues - `separate` is a new, unpopular and barely used tag. See http://taghistory.raifer.tech/#***/bin/yes&***/bin/no&***/bin/separate

At various times separate has been listed for shelter/bench and bin on various bits of the wiki, although I appreciate that seems to change every time I look at it currently. iD gives a tick box in it's main form, which populates yes/no, but in the drop down it offers yes/no/separate/unknown(!) presumably just every option with more than 5 uses or something.

> (though as mapper is unable to distinguish between `yes` and `separate` in SC it is not useful to do it)

I can distinguish if the bin is on the pole, i.e. it's part of the bus stop, or just happens to be nearby on the street, and so is a separate bin.

> > So if the bin is on the post itself, do you add the amenity=waste_basket tag to the highway=bus_stop node itself? Or put another node practically on top of it?
> 
> but in such case I would put node next to it.

Assuming the node is supposed to represent the actual bus stop position itself, then by adding it next to it you're actually giving it an inaccurate position.

I guess I'm a bit conceptually confused, either `highway=bus_stop` is a theoretical concept and should perhaps be an area or centred on the stop point, with the pole (with optional bin), shelter (with optional bench) and potentially standalone nearby bin all micro-mapped. Or the `highway=bus_stop` represents the stop (probably at the pole if present), and the shelter/bench/bin are properties of that stop and like the highway they may not be part of the stop but linked ones could be mapped separately (if people don't want to, or can't run a search in the local area).

> > and crucially if there isn't a amenity=waste_basket
> 
> In such case `bin=separate` is simply wrong (even if that makes counting for one specific scenario easier)

No, it just means some data is missing and needs populating, which a QC tool can capture and flag up, or a future version of SC could allow us to map the bin location.

> > https://wiki.openstreetmap.org/wiki/One_feature,_one_OSM_element
> 
> `bin=yes` is not marking a waste basket. It marks that something has associated waste basket. In particular, multiple objects with `bin=yes` may be associated with one waste basket and such waste basket may not be mapped.

Then why are we even tagging it at all? Why aren't we just concentrating on adding the bins themselves?

> Similarly to `bridge=yes` on road (mapping as a property) and `man_made=bridge` (maps a bridge).
> 
> Or `embankment=yes` on road or railway vs mapped `man_made=embankment`.

Assuming the `man_made=bridge` is an area, why the hell aren't we enriching this data as part of the API (if stuff really needs it), rather than double-tagging it?

> > guess
> 
> Note that it will be never eliminated - even if `bin=separate` will become accepted you will never know whether `yes` is in meaning "and separate waste basket object may be present" or in "and separate waste basket object is not present".

Over time, that should improve, as people could re-verify and improve them, but like all the tagging on OSM, garbage in, garbage out (pun intended)! :laughing: 

> To count waste baskets exactly one needs to map them with `amenity=waste_basket` and count that (though coverage of area by waste baskets is anyway more important than raw instance count). And use `bin=yes` without nearby `amenity=waste_basket` as indicator in QA tool.

> > I can tune my heuristic for
> 
> Note that this tagging scheme is already used and it is too late to redefine what `yes` value means.

As mentioned, with a rolling resurvey, you could improve things.

Perhaps a stronger reasoning for my logic, if the transport provider decide to move/remove a bus stop, if the shelter, bench and bin are part of the stop (e.g. bin on pole, combined shelter and bench) then you move/delete the node and it's all good. Whereas if the bin is separate, or it's actually a nearby park bench that's earned the bench=yes, or a seaside style bus shelter ( https://www.alamy.com/stock-photo-seaside-bus-stop-28263028.html ), those features are unlikely to be moved/removed, they just happen to be incidental to the stop but are _separate_ features, so should be mapped separately.

Or some busy bus stops have two independent shelters, whereas other busy areas have two bus stops next to each other, how do you show the second shelter if yes can mean it's nearby or it's part of the bus stop.

Incidentally, this wiki page suggests SC shouldn't show the quest if there's one nearby (well technically not for bin but for shelter/bench) "If there is a shelter that is not tagged separately with amenity=shelter":
https://wiki.openstreetmap.org/wiki/Tag:public_transport%3Dplatform

Which would seem like the sane way of using this tag (if separate wasn't used) but would be harder for SC. Separate seems like the good counter to that "rule".

Anyway, aside from #2354, in the short term, given the wiki says "Presence of a waste basket at/in a facility", the thing SC probably needs to hint at/OSM community needs to decide, is what counts as "present at"? Rather than just there happens to be a bus stop and a bin on the same street (the first two are probably some of the best):
https://www.google.com/maps/@51.5368599,-0.1011825,3a,75y,90.89h,81.63t/data=!3m6!1e1!3m4!1sxqD7sD4xOHhMl5NC67sFeg!2e0!7i16384!8i8192
https://www.google.com/maps/@51.5383155,-0.0993772,3a,75y,78.39h,78.09t/data=!3m6!1e1!3m4!1slqFHhM7cwpZ0XJCNHai2Ow!2e0!7i16384!8i8192
https://wiki.openstreetmap.org/wiki/File:Tactile_paving_bus_stop.jpg
http://rzhooker.com/busstop/thumbs
https://www.flickriver.com/photos/46898720@N07/35205771144/

What about if it's by the stop on the other side of the street, which seemed to be valid as yes during the shelter discussion?

**Comment by smichel17** [comment]
While this is an interesting discussion, it does seem to be off-topic for the SC issue tracker (though, imo, it would be a good idea if such conversations were accepted in the "Discussions" part of this GitHub project).

**Comment by matkoniecz** [comment]
> [iD] in the drop down it offers yes/no/separate/unknown(!) presumably just every option with more than 5 uses or something.

Yes, that is just the most used value

> >     (though as mapper is unable to distinguish between yes and separate in SC it is not useful to do it)

> I can distinguish if the bin is on the pole, i.e. it's part of the bus stop, or just happens to be nearby on the street, and so is a separate bin.

There is no established tagging scheme for that distinction. And `separate` value - even if would be accepted - refers to something else: whether bin is mapped separately, not whether it is tightly integrated with a bus stop.

Even if  `bin=separate` would be accepted then tagging `bin=separate` would be wrong if that bin is not yet mapped AND `bin=yes` would be correct if there is a bin next to bus stop not yet mapped. So it does not match your specific usecase anyway.

As there is no established tagging scheme for "is this specific bin attached to bus stop or next to it", SC is a bad place to invent and push it.

**Comment by matkoniecz** [comment]
> While this is an interesting discussion, it does seem to be off-topic for the SC issue tracker.

In this case there was report of a very large amount of notes generated due to possible mistake in SC - but turned out to be result of something else.

**Comment by westnordost** [comment]
Jesus, what is this huge discussion for?

Let's solve the issue as reported by harry-wg by simply changing the wording from "Does this bus stop have a waste basket" to "Is there a waste basket at this bus stop?".

**Comment by westnordost** [comment]
And I guess for any bus stop related quest: bench and shelter too

**Comment by peternewman** [comment]
> Let's solve the issue as reported by harry-wg by simply changing the wording from "Does this bus stop have a waste basket" to "Is there a waste basket at this bus stop?".

That certainly improves things for the adjacent ones thanks @westnordost .

I'd still say these two are more ambiguous and would perhaps result in a note (which might at least confirm if the bin is mapped):
https://www.google.com/maps/@51.5368599,-0.1011825,3a,75y,90.89h,81.63t/data=!3m6!1e1!3m4!1sxqD7sD4xOHhMl5NC67sFeg!2e0!7i16384!8i8192
https://www.google.com/maps/@51.5383155,-0.0993772,3a,75y,118.2h,76.18t/data=!3m6!1e1!3m4!1slqFHhM7cwpZ0XJCNHai2Ow!2e0!7i16384!8i8192
